src/Xrefcheck/Config/Default.hs

```diff
@@ -96,3 +123,20 @@ scanners:
         , "../../merge_requests"
         , "../../merge_requests/*"
         ]
+
+    externalRefRedirects :: Builder
+    externalRefRedirects = case flavor of
+      GitHub ->
+        [int||
+        - on: permanent
+          outcome: invalid|]
+      GitLab ->
+        [int||
+        - on: permanent
+          outcome: invalid
+        # GitLab redirects non-existing files to the repository's main page
+        # with a 302 code instead of answering with a 404 response.
+        - from: https?://gitlab.com/.*/-/blob/.*
+          to: https?://gitlab.com/.*
+          on: 302
+          outcome: invalid|]
```

Comment:
Very good :+1: 

The fact that those regexes are internally glob patterns works for us: somehow that `.*` in between does not match things that contain slashes, and this is exactly what we what.